File: Objective-C/TripServiceKata/TripServiceKata/UserSession.m
#import "UserSession.h"
#import "User.h"
#import "DependendClassCallDuringUnitTestException.h"


@implementation UserSession

+ (UserSession*)sharedInstance
{
    static UserSession *_sharedInstance = nil;
    static dispatch_once_t oncePredicate;
    
    dispatch_once(&oncePredicate, ^{
        _sharedInstance = [[UserSession alloc] init];
    });
    return _sharedInstance;
}

-(BOOL)isUserLoggedIn:(User*)user
{
    [DependendClassCallDuringUnitTestException raise:@"Unit Test Exception" format:@"UserSession.IsUserLoggedIn() should not be called in an unit test"];
    return nil;
}


-(User*)getLoggedUser
{
    [DependendClassCallDuringUnitTestException raise:@"Unit Test Exception" format:@"UserSession.getLoggedUser() should not be called in an unit test"];
    
    return nil;
}

@end

File: python-objects/UserSession.py
from DependendClassCallDuringUnitTestException import DependendClassCallDuringUnitTestException


class UserSession(object):
    def __new__(cls):
        if not hasattr(cls, '_instance'):
            cls._instance = super(UserSession, cls).__new__(cls)
        return cls._instance

    @staticmethod
    def get_instance():
        return UserSession()

    def is_user_logged_in(self, user):
        raise DependendClassCallDuringUnitTestException(
            "UserSession.isUserLoggedIn() should not be called in an unit test")

    def get_logged_user(self):
        raise DependendClassCallDuringUnitTestException(
            "UserSession.getLoggedUser() should not be called in an unit test")
